### ISSUE TITLE
Why are ray directions normalized by z?

### ISSUE THREAD

**Issue by bmanga** [princeton-computational-imaging/NSF#5]
Hi,

In the code I noticed the following line:
```python
ray_directions = ray_directions / ray_directions[:,2:3] # normalize by z
```

This is also mentioned in the paper, but I was wondering why that's the case.

Thanks :)

**Comment by Ilya-Muromets** [comment]
In `generate_ray_directions` that's mostly for convenience, so that you can find the intersection `t` of the ray and a plane at depth `d` as just `t = d`. 

There's also depth scaling in:
`torch.sum(intersection_points[:, :2] * (self.u0 / (torch.abs(termination) + 1e-6)), dim=1)`

Because you want to keep the projected points between `[0,1]` for the MLP encoding. Otherwise the farther rays go into space, the larger `uv` become.
